students/njschafi/Lesson03/assignment/test_basic_operations.py

```diff
@@ -0,0 +1,115 @@
+"""Unit test for basic operations program"""
+# pylint: disable=W0614, C0103, W0401
+from unittest import TestCase
+from customer_model import *
+from basic_operations import add_customer, search_customer, delete_customer
+from basic_operations import update_customer_credit, list_active_customers
+
+# create a set of customers
+user_1 = {'customer_id': '1234', 'name': 'Post', 'lastname': 'Malone',
+          'home_address': '456 Manure ave', 'phone_number': '0293939393',
+          'email_address': 'postymaloney@gmail.com', 'status': True,
+          'credit_limit': 1000.00}
+
+user_2 = {'customer_id': '546', 'name': 'Howard', 'lastname': 'Moon',
+          'home_address': '789 Satsuma way', 'phone_number': '014445555',
+          'email_address': 'howardmoon@gmail.com', 'status': False,
+          'credit_limit': 5000.00}
+
+user_3 = {'customer_id': '7899', 'name': 'Vince', 'lastname': 'Noir',
+          'home_address': '2234 Flannel PL', 'phone_number': '096667777',
+          'email_address': 'vincenoir@gmail.com', 'status': True,
+          'credit_limit': 99999.56}
+
+
+def drop_db():
+    """Drops customer table"""
+    database.drop_tables([Customer])
+    database.close()
+
+
+def create_empty_db():
+    """Creates an empty customer database"""
+    drop_db()
+    database.create_tables([Customer])
+    database.close()
+
+
+
+class BasicOperationsTest(TestCase):
+    """Tests basic_operations program, along with customer_model"""
+    def test_add_customer(self):
+        """Tests if a new customer is added to database"""
+        create_empty_db()
+        add_customer(**user_1)
+        query = Customer.get(Customer.customer_id == user_1['customer_id'])
+        self.assertEqual(user_1['name'], query.customer_name)
+        self.assertEqual(user_1['lastname'], query.customer_last_name)
+        self.assertEqual(user_1['home_address'], query.customer_address)
+        self.assertEqual(user_1['phone_number'], query.customer_phone)
+        self.assertEqual(user_1['email_address'], query.customer_email)
+        self.assertEqual(user_1['status'], query.customer_status)
+        self.assertEqual(user_1['credit_limit'], query.customer_limit)
+
+        # add another person
+        add_customer(**user_2)
+        query = Customer.get(Customer.customer_id == user_2['customer_id'])
+        self.assertEqual(user_2['name'], query.customer_name)
+        self.assertEqual(user_2['lastname'], query.customer_last_name)
+        self.assertEqual(user_2['home_address'], query.customer_address)
+        self.assertEqual(user_2['phone_number'], query.customer_phone)
+        self.assertEqual(user_2['email_address'], query.customer_email)
+        self.assertEqual(user_2['status'], query.customer_status)
+        self.assertEqual(user_2['credit_limit'], query.customer_limit)
+
+        # add a duplicate person
+        self.assertRaises(ValueError, add_customer(**user_2))

```

Comment:
Doing _assertRaises_ will cause your test to fail, since the exception is not contained. Instead, do it as a context manager:

        with self.assertRaises(ValueError):
            add_customer(**user_2) 

Your test will verify the correct exception is raised, but the exception itself will be "trapped" by the context manager.